### PR TITLE
Bugfix for loopCount

### PR DESCRIPTION
**Pull request by LululuSir**
### New Pull Request Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar pull request in the [project](https://github.com/rs/SDWebImage/pulls) and found none

* [ ] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [ ] I have added the required tests to prove the fix/feature I am adding
* [ ] I have updated the documentation (if necessary)
* [ ] I have run the tests and they pass
* [ ] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / reffers to the following issues: ...

### Pull Request Description

...

### CODE REVIEW

**Comment by LululuSir** [comment]
FLAnimated has same issue, and I pull a request too:
https://github.com/Flipboard/FLAnimatedImage/pull/198/files

**Comment by dreampiggy** [comment]
😅 It seems that we do not follow the `Correct standard`, but the `De facto standard`. See here about that GIF loop count behavior:
https://bugs.chromium.org/p/chromium/issues/detail?id=592735

This also happended for WebP, such as the code in `SDWebImageWebPCoder.m`:

> WebP standard says 0 duration is used for canvas updating but not showing image, but actually Chrome and other implementations set it to 100ms if duration is lower or equal than 10ms
Some animated WebP images also created without duration, we should keep compatibility

**Comment by LululuSir** [comment]
Thanks for the loop count behavior info. And in our team, the Designer and H5 Engineer use Chrome far more than Safari, so we will use the Chrome standard.

File: SDWebImage/SDWebImageGIFCoder.m [hunk 1]
```diff
@@ -72,13 +72,15 @@ - (UIImage *)decodedImageWithData:(NSData *)data {
...
         
-        NSUInteger loopCount = 0;
+        NSUInteger loopCount = 1;
         NSDictionary *imageProperties = (__bridge_transfer NSDictionary *)CGImageSourceCopyProperties(source, nil);
         NSDictionary *gifProperties = [imageProperties valueForKey:(__bridge_transfer NSString *)kCGImagePropertyGIFDictionary];
         if (gifProperties) {
             NSNumber *gifLoopCount = [gifProperties valueForKey:(__bridge_transfer NSString *)kCGImagePropertyGIFLoopCount];
-            if (gifLoopCount != nil) {
-                loopCount = gifLoopCount.unsignedIntegerValue;
+            if (gifLoopCount != nil && gifLoopCount.unsignedIntegerValue == 0) {
```
Comment (dreampiggy) [inline SDWebImage/SDWebImageGIFCoder.m]: @LululuSir This `If has no loopCount info, loop count would be 1` seems adoptable. But what about these change ? Should we add loop count by 1 ?

File: SDWebImage/SDWebImageGIFCoder.m [hunk 2]
```diff
@@ -72,13 +72,15 @@ - (UIImage *)decodedImageWithData:(NSData *)data {
...
+                loopCount = 0;
+            } else {
+                loopCount = gifLoopCount.unsignedIntegerValue + 1;
```
Comment (dreampiggy) [inline SDWebImage/SDWebImageGIFCoder.m]: @LululuSir This If has no loopCount info, loop count would be 1 seems adoptable. But what about these change ? Should we add loop count by 1 ?

**Comment by dreampiggy** [comment]
What about this test GIF. The `loop count` value is 2, but Chrome seems show animation 3 times, Safari 2 times...Our lib show 2 times 😅 
full2loop

**Comment by dreampiggy** [comment]
It seems another cross-browser compatibility issue, like APNG loop count :)

My blog point this out (Chinese): http://dreampiggy.com/2017/03/06/%E5%AE%A2%E6%88%B7%E7%AB%AF%E4%B8%8A%E5%8A%A8%E6%80%81%E5%9B%BE%E6%A0%BC%E5%BC%8F%E5%AF%B9%E6%AF%94%E5%92%8C%E8%A7%A3%E5%86%B3%E6%96%B9%E6%A1%88/

**Comment by dreampiggy** [comment]
And I want to point out that since we support custom coder, you do not need to wait for SD new version release. Just subclass `SDWebImageGIFCoder` and override that method for your behavior.


```objective-c
@interface MySDWebImageGIFCoder : SDWebImageGIFCoder

@end

@implementation MySDWebImageGIFCoder

- (UIImage *)decodedImageWithData:(NSData *)data {
    // copy the code
    // override the loop count calculation
    return image;
}

@end

// Our GIF coder not add to default coders array in 4.x version(But in 5.x we will because of SDAnimatedImageView)
 [[SDWebImageCodersManager sharedInstance] addCoder:[MySDWebImageGIFCoder sharedCoder]];
```

**Comment by dreampiggy** [comment]
In my opinion, I don't agree the behavior about that `Animation count` = `loop count` + 1, this will cause a big issue:

How should I loop a image with only once ?

For GIF, you can leave without any loop information, so the GIF animation plays once.

But however, what about other image format designed ? See WebP's bug:
https://bugs.chromium.org/p/chromium/issues/detail?id=649264

Due to Google change the behavior to make it the same as GIF, you **CAN NOT** create a animated webp with only 1 loop count. Because WebP standard says if you not specify loop count, loop infinity. It's really stupid.

So, I agree the behavior that GIF without loop count should play once, but whether we should plus 1 to loop count is under consideration.

**Comment by LululuSir** [comment]
Yes, I think you don't have to add 1 loop count, it could consistent with safari. And I will change it just in our inner version.
PS: Can I have you contact information, please? My weChat: Q_lu0330

**Pull request merged**